virtual_machines.go

```diff
@@ -204,13 +207,18 @@ func (v *VirtualMachine) UnlinkDisk(diskID string, force bool) (task *Task, err
 	return NewTask(upid, v.client), nil
 }
 
-func (v *VirtualMachine) MoveDisk(disk, storage string) (task *Task, err error) {
+func (v *VirtualMachine) MoveDisk(disk string, params *VirtualMachineMoveDiskOptions) (task *Task, err error) {
 	var upid UPID
 
-	err = v.client.Post(fmt.Sprintf("/nodes/%s/qemu/%d/move_disk", v.Node, v.VMID), map[string]string{
-		"disk":    disk,
-		"storage": storage,
-	}, &upid)
+	if params == nil {
+		params = &VirtualMachineMoveDiskOptions{}
+	}
+
+	if disk != "" {
```

Comment:
if disk is empty should we error? it can't be empty ever can it?

Reply:
ya that's fair, the disk being empty will raise a proxmox error so just bubble that